src/pages/suggestions/edit/[suggestion].tsx

```diff
@@ -0,0 +1,185 @@
+        redirect: {
+          destination: "/403",
+          permanent: false,
+        },
+      }
+
+    return { props: { suggestion: data } }
+  } catch {
+    return { notFound: true }
+  }
+}
+
+interface EditSuggestionPageProps {
+  suggestion: Song
+}
+
+const EditSuggestionPage = (props: EditSuggestionPageProps) => {
+  const supabase = useSupabaseClient<Database>()
+  const router = useRouter()
+  const dispatch = useDispatch()
+
+  const lastEditedUuid = useSelector((state: AppState) => state.editSuggestion.lastEditedUuid)
+  const suggestion = useSelector((state: AppState) => state.editSuggestion.suggestion)
+  const activeArea = useSelector((state: AppState) => state.editSuggestion.activeArea)
+  const deletedInstruments = useSelector(
+    (state: AppState) => state.editSuggestion.deletedInstrumentUuids
+  )
+
+  //If the user visits a new edit page, clear the previously edited from redux
+  if (props.suggestion.id !== lastEditedUuid) {
+    const suggestionInstruments: NewSuggestionInstrument[] = []
+    props.suggestion.song_instruments.forEach(
+      (element: { id: any; description: any; instrument: any }) => {
+        suggestionInstruments.push({
+          id: element.id,
+          description: element.description ?? "",
+          instrument: element.instrument,
+        })
+      }
+    )
+
+    dispatch(
+      updateEditSuggestion({
+        artist: props.suggestion.artist,
+        link: props.suggestion.link,
+        motivation: props.suggestion.motivation,
+        title: props.suggestion.title,
+        instruments: suggestionInstruments,
+      })
+    )
+
+    dispatch(updateDeletedInstrumentUuid([]))
+    dispatch(updateLastEditedUuid(props.suggestion.id))
+  }
+
+  const saveSuggestion = async (onSuccess: () => void, onError: () => void) => {
+    try {
+      const suggestionResponse = await updateSuggestion(supabase, suggestion, lastEditedUuid ?? "")
+      if (suggestionResponse.error) throw new Error("Failed to update song")
+
+      const suggestionId = suggestionResponse.data.at(0)!.id
+      const suggestionInstrumentResponse = await updateSuggestionInstruments(
+        supabase,
+        mapEditInstruments(suggestion, suggestionId)
+      )
+      if (suggestionInstrumentResponse.error) throw Error("Failed to update song_instruments")
+
+      const deleteSuggestionResponse = await deleteSuggestionInstruments(
+        supabase,
+        deletedInstruments
+      )
+      if (deleteSuggestionResponse.error) throw new Error("Failed to delete song_instruments")
+
+      await router.push(`/suggestions/${suggestionId}`).then(() => {
+        onSuccess()
+        dispatch(setActiveArea(Area.SongInformation))
+        dispatch(updateLastEditedUuid(""))
+        dispatch(updateDeletedInstrumentUuid([]))
+      })
+    } catch (error) {
+      onError()
+    }
+  }
+
+  const callSaveSuggestion = (success: () => void, error: () => void) => {
+    saveSuggestion(success, error).catch(() => {})
+  }
```

Comment:
Why?

Reply:
Sonarcloud was unhappy with feeding the method into the type defined in the interface. Sonarcloud is overly stingy often.